runtime/recipe/particle.js

```diff
@@ -86,6 +86,11 @@ class Particle {
   isResolved() {
     assert(Object.isFrozen(this));
     // TODO: slots
+    if (this.consumedSlotConnections.length > 0) {
+      let fulfilledSlotConnections = this.consumedSlotConnections.filter(connection => connection.targetSlot !== undefined);
```

Comment:
A connection that isn't fulfilled can be resolved, as long as it isn't a required connection